frost-ed448/src/lib.rs

```diff
@@ -0,0 +1,405 @@
+#![allow(non_snake_case)]
+#![deny(missing_docs)]
+#![doc = include_str!("../README.md")]
+
+use ed448_goldilocks::{
+    curve::{edwards::CompressedEdwardsY, ExtendedPoint},
+    Scalar,
+};
+use rand_core::{CryptoRng, RngCore};
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+use frost_core::{frost, Ciphersuite, Field, Group};
+
+#[cfg(test)]
+mod tests;
+
+pub use frost_core::Error;
+
+#[derive(Clone, Copy)]
+/// An implementation of the FROST(Ed448, SHAKE256) ciphersuite scalar field.
+pub struct Ed448ScalarField;
+
+impl Field for Ed448ScalarField {
+    type Scalar = Scalar;
+
+    type Serialization = [u8; 57];
+
+    fn zero() -> Self::Scalar {
+        Scalar::zero()
+    }
+
+    fn one() -> Self::Scalar {
+        Scalar::one()
+    }
+
+    fn invert(scalar: &Self::Scalar) -> Result<Self::Scalar, Error> {
+        if *scalar == <Self as Field>::zero() {
+            Err(Error::InvalidZeroScalar)
+        } else {
+            Ok(scalar.invert())
+        }
+    }
+
+    fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self::Scalar {
+        Scalar::random(rng)
+    }
+
+    fn serialize(scalar: &Self::Scalar) -> Self::Serialization {
+        let bytes = scalar.to_bytes();
+        std::array::from_fn(|i| if i < 56 { bytes[i] } else { 0 })
+    }
+
+    fn deserialize(buf: &Self::Serialization) -> Result<Self::Scalar, Error> {
+        match Scalar::from_canonical_bytes(*buf) {
+            Some(s) => Ok(s),
+            None => Err(Error::MalformedScalar),
+        }
+    }
+
+    fn little_endian_serialize(scalar: &Self::Scalar) -> Self::Serialization {
+        Self::serialize(scalar)
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+/// An implementation of the FROST(Ed448, SHAKE256) ciphersuite group.
+pub struct Ed448Group;
+
+impl Group for Ed448Group {
+    type Field = Ed448ScalarField;
+
+    type Element = ExtendedPoint;
+
+    type Serialization = [u8; 57];
+
+    fn cofactor() -> <Self::Field as Field>::Scalar {
+        Scalar::one()
+    }
+
+    fn identity() -> Self::Element {
+        Self::Element::identity()
+    }
+
+    fn generator() -> Self::Element {
+        Self::Element::generator()
+    }
+
+    fn serialize(element: &Self::Element) -> Self::Serialization {
+        element.compress().0
```

Comment:
Funny they don't have a proper means of getting these bytes out

Reply:
To be fair `curve25519-dalek`, while it has `to/as_bytes`, also makes the bytes `pub`.

I want to eventually try to make the `Ed448-Goldilocks` API as similar as `curve25519-dalek` as possible